From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jianfeng Liu <Jianfeng.Liu@cixtech.com>
Date: Tue, 17 Mar 2026 17:29:25 +0800
Subject: i2c: add acpi support for cadence driver

---
 drivers/i2c/busses/i2c-cadence.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/i2c/busses/i2c-cadence.c b/drivers/i2c/busses/i2c-cadence.c
index 111111111111..222222222222 100644
--- a/drivers/i2c/busses/i2c-cadence.c
+++ b/drivers/i2c/busses/i2c-cadence.c
@@ -5,6 +5,7 @@
  * Copyright (C) 2009 - 2014 Xilinx, Inc.
  */
 
+#include <linux/acpi.h>
 #include <linux/clk.h>
 #include <linux/delay.h>
 #include <linux/i2c.h>
@@ -1449,6 +1450,14 @@ static const struct of_device_id cdns_i2c_of_match[] = {
 };
 MODULE_DEVICE_TABLE(of, cdns_i2c_of_match);
 
+#ifdef CONFIG_ACPI
+static const struct acpi_device_id cdns_i2c_acpi_match[] = {
+	{ "CIXH200B", 0 },
+	{ }
+};
+MODULE_DEVICE_TABLE(acpi, cdns_i2c_acpi_match);
+#endif
+
 /**
  * cdns_i2c_detect_transfer_size - Detect the maximum transfer size supported
  * @id: Device private data structure
@@ -1639,6 +1648,7 @@ static struct platform_driver cdns_i2c_drv = {
 	.driver = {
 		.name  = DRIVER_NAME,
 		.of_match_table = cdns_i2c_of_match,
+		.acpi_match_table = ACPI_PTR(cdns_i2c_acpi_match),
 		.pm = &cdns_i2c_dev_pm_ops,
 	},
 	.probe  = cdns_i2c_probe,
-- 
Cixtech

